dispatch.go

```diff
@@ -40,13 +40,20 @@ func (d *dispatch) Call(ctx context.Context, name string, arg interface{}, res i
 	profiler := d.log.StartProfiler("call %s", name)
 	defer profiler.Stop()
 
+	select {
```

Comment:
Hmm...why is this needed here?

Actually, I'm a bit worried that the body of Encode doesn't wait for <-d.stopCh, too. is doneCh closed when stopCh is?